packages/core/src/render3/component.ts

```diff
@@ -165,7 +165,7 @@ export function detectChanges<T>(component: T) {
   ngDevMode && assertNotNull(hostNode.data, 'hostNode.data');
   const oldView = enterView(hostNode.view !, hostNode);
   try {
-    (component.constructor as ComponentType<T>).ngComponentDef.r(0, 0);
+    (component.constructor as ComponentType<T>).ngComponentDef.r(1, 0);
```

Comment:
Comment that explains why the indices 1 and zero are used here?

(would be a ice bonus to also add a comment that says what `r` is/does)